cli/src/api/mod.rs

```diff
@@ -63,34 +65,36 @@ enum APIResult<T> {
 }
 
 impl PhylumApi {
-    async fn get<T: serde::de::DeserializeOwned>(&self, path: String) -> Result<T> {
-        let body = self.client.get(path).send().await?.text().await?;
+    async fn get<T: DeserializeOwned>(&self, path: String) -> Result<T> {
+        self.send_request::<_, ()>(Method::GET, path, None).await
+    }
 
-        let api_obj = serde_json::from_str::<APIResult<T>>(&body)
-            .map_err(|e| PhylumApiError::Other(e.into()))?;
-        match api_obj {
-            APIResult::Ok(api_obj) => Ok(api_obj),
-            APIResult::Err { msg } => Err(PhylumApiError::Other(anyhow::anyhow!(msg))),
-        }
+    async fn put<T: DeserializeOwned, S: serde::Serialize>(&self, path: String, s: S) -> Result<T> {
+        self.send_request(Method::PUT, path, Some(s)).await
+    }
+
+    async fn delete<T: DeserializeOwned>(&self, path: String) -> Result<T> {
+        self.send_request::<_, ()>(Method::DELETE, path, None).await
     }
 
-    async fn put<T: serde::de::DeserializeOwned, S: serde::Serialize>(
+    async fn send_request<T: DeserializeOwned, B: Serialize>(
         &self,
+        method: Method,
         path: String,
-        s: S,
+        body: Option<B>,
     ) -> Result<T> {
-        let body = self.client.put(path).json(&s).send().await?.text().await?;
-
-        let api_obj = serde_json::from_str::<APIResult<T>>(&body)
-            .map_err(|e| PhylumApiError::Other(e.into()))?;
-        match api_obj {
-            APIResult::Ok(api_obj) => Ok(api_obj),
-            APIResult::Err { msg } => Err(PhylumApiError::Other(anyhow::anyhow!(msg))),
+        let mut request = self.client.request(method, path);
+        if let Some(body) = body {
+            request = request.json(&body);
         }
-    }
 
-    async fn delete<T: serde::de::DeserializeOwned>(&self, path: String) -> Result<T> {
-        let body = self.client.delete(path).send().await?.text().await?;
+        let response = request.send().await?;
+        let success = response.status().is_success();
+        let body = response.text().await?;
+
+        if !success {
+            return Err(anyhow!(body).into());
```

Comment:
It would be nice if this error also included the HTTP status code. Maybe as context?

```
    .with_context(|| format!("HTTP status error ({})", status))
```

Reply:
You can't provide a context since this is not an anyhow result. That's why I brought the question of returning anyhow results up to you in Slack.